src/java.base/share/classes/java/net/Socket.java

```diff
@@ -616,10 +613,15 @@ void setConnected() {
      *        cause it to wakeup and close the socket. This method will then throw
      *        {@code SocketException} with the interrupt status set.
      * </ol>
+     * <p> Besides above noted cases, the socket will also be closed when a
+     * connection attempt fails with an {@link IOException}.
```

Comment:
I think it would be better to put the new paragraph before the paragraph on interrupt. I think we'll end up with something like " If the endpoint is an unresolved InetSocketAddress, the connection cannot be established, or the timeout expires before the connection is established, then the Socket is closed and IOException is thrown".

Reply:
Implemented the requested change in 7e633d8d213a950304c550f93a1697c1f2530a05. Note that I excluded the  timeout-related part for `connect(SocketAddress)`, since it doesn't receive timeout as input. Yet, cloned this changes to `connect(SocketAddress,int)` and retained the timeout-related part. Let me know if you want me to add the timeout-related part to the `connect(SocketAddress)` Javadoc too.